homeassistant/components/balboa/climate.py

```diff
@@ -0,0 +1,188 @@
+"""Support for Balboa Spa Wifi adaptor."""
+from __future__ import annotations
+
+import math
+
+from homeassistant.components.climate import ClimateEntity
+from homeassistant.components.climate.const import (
+    CURRENT_HVAC_HEAT,
+    CURRENT_HVAC_IDLE,
+    FAN_OFF,
+    HVAC_MODE_AUTO,
+    HVAC_MODE_HEAT,
+    SUPPORT_FAN_MODE,
+    SUPPORT_PRESET_MODE,
+    SUPPORT_TARGET_TEMPERATURE,
+)
+from homeassistant.const import (
+    ATTR_TEMPERATURE,
+    PRECISION_HALVES,
+    PRECISION_WHOLE,
+    TEMP_CELSIUS,
+    TEMP_FAHRENHEIT,
+)
+
+from .balboa_entity import BalboaEntity
+from .const import CLIMATE, CLIMATE_SUPPORTED_FANSTATES, CLIMATE_SUPPORTED_MODES, DOMAIN
+
+
+async def async_setup_entry(hass, entry, async_add_entities):
+    """Set up the spa climate device."""
+    async_add_entities(
+        [
+            BalboaSpaClimate(
+                hass,
+                entry,
+                hass.data[DOMAIN][entry.entry_id],
+                CLIMATE,
+            )
+        ],
+        True,
```

Comment:
```suggestion
```

`async_update` isn't implemented but `True` will ask HA to update the entity before adding it